Library/ViewModels/ManagePledgeSummaryViewViewModel.swift

```diff
@@ -0,0 +1,136 @@
+
+    self.pledgeAmountText = projectAndBacking
+      .map { attributedCurrency(with: $0.0, amount: $0.1.amount) }
+      .skipNil()
+
+    let shippingAmount = backing
+      .map { Double($0.shippingAmount ?? 0) }
+
+    self.shippingAmountText = self.projectSignal
+      .combineLatest(with: shippingAmount)
+      .map { shippingValue(with: $0.0, with: $0.1) }
+      .skipNil()
+
+    self.totalAmountText = projectAndBacking
+      .combineLatest(with: shippingAmount)
+      .map(unpack)
+      .map { project, backing, shippingAmount in
+        attributedCurrency(with: project, amount: backing.amount + shippingAmount)
+      }
+      .skipNil()
+
+    self.shippingLocationText = backing.ignoreValues()
+      .map { "Shipping: Australia" }
+
+    self.shippingLocationStackViewIsHidden = self.projectSignal
+      .map(shouldHideShippingLocationStackView)
+  }
+
+  private let (projectSignal, projectObserver) = Signal<Project, Never>.pipe()
+  public func configureWith(_ project: Project) {
+    self.projectObserver.send(value: project)
+  }
+
+  public let backerNumberText: Signal<String, Never>
+  public let backingDateText: Signal<String, Never>
+  public let pledgeAmountText: Signal<NSAttributedString, Never>
+  public let shippingAmountText: Signal<NSAttributedString, Never>
+  public let shippingLocationStackViewIsHidden: Signal<Bool, Never>
+  public let shippingLocationText: Signal<String, Never>
+  public let totalAmountText: Signal<NSAttributedString, Never>
+
+  public var inputs: ManagePledgeSummaryViewViewModelInputs { return self }
+  public var outputs: ManagePledgeSummaryViewViewModelOutputs { return self }
+}
+
+private func shouldHideShippingLocationStackView(_ project: Project) -> Bool {
+  guard let backing = project.personalization.backing,
+    let _ = backing.rewardId else {
+    return true
+  }
+  if let reward = backing.reward {
+    return !reward.shipping.enabled || reward.isNoReward
+  }
+
+  return backing.locationId.isNil
+}
+
+private func formattedPledgeDate(_ backing: Backing) -> String {
+  let formattedDate = Format.date(secondsInUTC: backing.pledgedAt, dateStyle: .long, timeStyle: .none)
+  return "As of \(formattedDate)"
+}
+
+private func attributedCurrency(with project: Project, amount: Double) -> NSAttributedString? {
+  let defaultAttributes = checkoutCurrencyDefaultAttributes()
+    .withAllValuesFrom([.foregroundColor: UIColor.ksr_green_500])
+  let superscriptAttributes = checkoutCurrencySuperscriptAttributes()
+  guard
+    let attributedCurrency = Format.attributedCurrency(
+      amount,
+      country: project.country,
+      omitCurrencyCode: project.stats.omitUSCurrencyCode,
+      defaultAttributes: defaultAttributes,
+      superscriptAttributes: superscriptAttributes
+    ) else { return nil }
+
+  let combinedAttributes = defaultAttributes
+    .withAllValuesFrom(superscriptAttributes)
+
+  return Format.attributedAmount(attributes: combinedAttributes) + attributedCurrency
```

Comment:
Seeing as we don't need the `+` sign, we can just do:

```swift
private func attributedCurrency(with project: Project, amount: Double) -> NSAttributedString? {
  let defaultAttributes = checkoutCurrencyDefaultAttributes()
    .withAllValuesFrom([.foregroundColor: UIColor.ksr_green_500])
  let superscriptAttributes = checkoutCurrencySuperscriptAttributes()
  guard
    let attributedCurrency = Format.attributedCurrency(
      amount,
      country: project.country,
      omitCurrencyCode: project.stats.omitUSCurrencyCode,
      defaultAttributes: defaultAttributes,
      superscriptAttributes: superscriptAttributes
    ) else { return nil }

  return attributedCurrency
}
```

Reply:
Much nicer than my solution :)